packages/service-library/src/servicelib/rabbitmq/_client.py

```diff
@@ -20,6 +21,63 @@
 
 _DEFAULT_PREFETCH_VALUE: Final[int] = 10
 _DEFAULT_RABBITMQ_EXECUTION_TIMEOUT_S: Final[int] = 5
+_HEADER_X_DEATH: Final[str] = "x-death"
+
+DEFAULT_UNEXPECTED_ERROR_RETRY_DELAY_S: Final[float] = 1
+DEFAULT_UNEXPECTED_ERROR_MAX_ATTEMPTS: Final[NonNegativeInt] = 15
+
+
+def _get_delayed_exchange_name(exchange_name: str) -> str:
+    return f"delayed_{exchange_name}"
+
+
+def _get_x_death_count(message: aio_pika.abc.AbstractIncomingMessage) -> int:
+    count: int = 0
+
+    x_death: list[dict[str, Any]] = message.headers.get(_HEADER_X_DEATH, [])
+    if x_death:
+        count = x_death[0]["count"]
+
+    return count
```

Comment:
how does that pass linting? you return int | Any here no?

Reply:
Well no, the type is inferred from the first line. I know it looks like "Any" is assigned, but I think that the tooling treats it as int when you assign Any since it could also may be an int. I'm not sure how it works.